src/main/java/core/basesyntax/operation/ShopServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.transaction.FruitTransaction;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+    private Map<String, Integer> fruits = new HashMap<>();
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void addFruits(String fruit, int quantity) {
+        fruits.put(fruit, fruits.getOrDefault(fruit, 0) + quantity);
+    }
+
+    @Override
+    public int getQuantity(String fruit) {
+        return fruits.getOrDefault(fruit, 0);
+    }
+
+    @Override
+    public Map<String, Integer> getFruits() {
+        return fruits;
+    }

```

Comment:
According to the principle of encapsulation, it's not a good practice to return a mutable object. This may lead to the situation where the `fruits` map can be modified outside of the `ShopServiceImpl` class. Consider returning a copy of the map or an unmodifiable view of the map.